File: src/cpp/src/rosenbrock.cpp
#include "rosenbrock.h"

double rosenbrock(vector<double> x){
    double y=0;

    for(int i=0; i<int(x.size())-1; i++){
        y+=100*pow((x[i+1]-pow(x[i], 2)), 2)+pow((x[i]-1), 2);
    }
    return y;
}

File: src/matlab/benchmark_functions/rosenbrock.m
%% Rosenbrock function implementation
%% Author:  Pietro Mosca
%% Email:   pietromosca1994@gmail.com
%% Date:    20.09.2020
%% Reference: https://en.wikipedia.org/wiki/Rosenbrock_function

function [y]=rosenbrock(x)
d = length(x);
sum = 0;
for i = 1:(d-1)
	xi = x(i);
	xnext = x(i+1);
	new = 100*(xnext-xi^2)^2 + (xi-1)^2;
	sum = sum + new;
end
y = sum;
end